starknet-replay/src/runner/replay_state_reader.rs

```diff
@@ -0,0 +1,116 @@
+//! This module implements [`blockifier::state::state_api::StateReader`] for use
+//! in Starknet transactions replay. The calls to read the blockchain state rely
+//! on the Starknet RPC protocol.
+
+use blockifier::execution::contract_class::ContractClass as BlockifierContractClass;
+use blockifier::state::errors::StateError;
+use blockifier::state::state_api::{StateReader, StateResult};
+use starknet::core::types::ContractClass as StarknetContractClass;
+use starknet_api::core::{ClassHash, CompiledClassHash, ContractAddress, Nonce};
+use starknet_api::hash::StarkFelt;
+use starknet_api::state::StorageKey;
+
+use crate::block_number::BlockNumber;
+use crate::runner::replay_class_hash::ReplayClassHash;
+use crate::storage::rpc::{contract_class, RpcStorage};
+
+pub struct ReplayStateReader<'a> {
+    /// The reference to [`crate::storage::rpc::RpcStorage`] to make RPC calls.
+    storage: &'a RpcStorage,
+
+    /// The block number used to query the state.
+    block_number: BlockNumber,
+}
+impl ReplayStateReader<'_> {
+    /// Constructs a new [`ReplayStateReader`] object.
+    ///
+    /// # Arguments
+    ///
+    /// - `storage`: The reference object exposing RPC calls to query blockchain
```

Comment:
You don't need to say "reference object". The type is part of the documentation, and it is clear that it is a reference.